thread/thread.cpp

```diff
@@ -648,42 +645,38 @@ namespace photon
         to->get_vcpu()->switch_count++;
     }
 
-    static void _photon_thread_die(thread* th) asm("_photon_thread_die");
-
 #if defined(__x86_64__)
 #if !defined(_WIN64)
     asm(
 DEF_ASM_FUNC(_photon_switch_context) // (void** rdi_to, void** rsi_from)
 R"(
+        rdfsbase %rax
         push    %rbp
+        push    %rax
         mov     %rsp, (%rsi)
         mov     (%rdi), %rsp
+        pop     %rax
         pop     %rbp
+        wrfsbase %rax
         ret
 )"
 
 DEF_ASM_FUNC(_photon_switch_context_defer) // (void* rdi_arg, void (*rsi_defer)(void*), void** rdx_to, void** rcx_from)
 R"(
+        rdfsbase %rax
         push    %rbp
+        push    %rax
         mov     %rsp, (%rcx)
 )"
 
 DEF_ASM_FUNC(_photon_switch_context_defer_die) // (void* rdi_arg, void (*rsi_defer)(void*), void** rdx_to_th)
 R"(
         mov     (%rdx), %rsp
+        pop     %rax
         pop     %rbp
+        wrfsbase %rax
         jmp     *%rsi
 )"
-
-DEF_ASM_FUNC(_photon_thread_stub)
```

Comment:
_photon_thread_stub() becomes more complex and would better be implemented in C++.